packages/odyssey-react/src/components/Status/Status.tsx

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { FunctionComponent } from 'react'
+import classNames from "classnames";
+
+export type StatusVariants = 'neutral' | 'success' | 'caution' | 'danger';
+export type StatusProps = {
+  /**
+   * The status label.
+   */
+  label: string,
+  
+  /**
+   * Visually hides the status label.
+   */
+  labelHidden?: boolean,
+
+  /**
+   * The description of the present state/status.
+   */
+  descriptor: string,
+
+  /**
+   * aria-role to be applied to the status element when it's descriptor is expected to change asynchronously.
+   */
+  role?: 'status',
+
+  /**
+   * The visual variant to be displayed to the user.
+   * @default neutral
+   */
+  variant?: StatusVariants
+}
+
+/**
+ * Status is used to inform users by providing feedback on system states. Status can display broad 
+ * operational states as well as granular states like user status.
+ * 
+ * @component
+ * @todo [OKTA-398175](https://oktainc.atlassian.net/browse/OKTA-398175) - Move away from definition 
+ * lists Status component HTML
+ * @example <Status label={label} descriptor={descriptor} />
+ */
+const Status: FunctionComponent<StatusProps> = (props) => {
+  const { 
+    label,
+    descriptor,
+    labelHidden = false,
```

Comment:
This is not the API we want here. We can't control every permutation in our components using boolean props like this because it will not scale.

We need to have the ability to use composition of smaller modular parts. Consumers should be able to provide labels that can manage their own visibility. Label should be required in this API but should also by polymorphic in that it can be a simple string, or an entire react component that can be visible or hidden.

Reply:
Consider cases where composition doesn't ensure proper behavior. For instance, if a hidden label would require the addition of `aria-describedby` in this component or a separate visual element to be included as an affordance. In other words, `labelHidden` does not always mean "a hidden variant of label"; it may mean "a variant of Status itself where the label is hidden".

IMO we should evaluate these implementations on a case-by-case basis instead of making a blanket call that this sort of API is invalid.

Reply:
@edburyenegren-okta hmmm I believe the issue you're driving at is something we'd solve for using the power of JS composition. For an example, look at my work on `<Tooltip />`. Composition doesn't have to mean that we lose all ability to ensure successful a11y outcomes. I'm not advocating for composition in every case either, variants or other simple props will still be a large part of what we're building. But when we consider the fact that text will be necessary in isolation, and screen reader only content will be necessary in isolation it starts to naturally point toward an API that accepts composed components here and doesn't require us as library authors to maintain a larger API surface area.